src/OpenTracing/SpanOptions.php

```diff
@@ -43,7 +43,17 @@ public static function create(array $options)
 
                 case 'tags':
                     foreach ($value as $tag => $tagValue) {
-                        $spanOptions->tags[] = Tag::create($tag, $tagValue);
+                        if ($tag !== (string) $tag) {
+                            throw InvalidSpanOption::invalidTag($tag);
+                        }
+
+                        if (!is_scalar($tagValue)
+                            && !(is_object($tagValue) && method_exists($tagValue, '__toString'))
```

Comment:
As I said I am not sure about the constraints but regarding the class actually we have more validations:
- https://github.com/jcchavezs/opentracing-php/pull/30/files/778218235bfaa06eb27bc75899ef38d45fc9a712#diff-21ba99ce57fc3e03c5e5e2d89db4b14dR46
- https://github.com/jcchavezs/opentracing-php/pull/30/files/778218235bfaa06eb27bc75899ef38d45fc9a712#diff-21ba99ce57fc3e03c5e5e2d89db4b14dR61

In any case, for me the benefits of this `SpanOptions` is that:
1. Allows you to pass references in a convenient way allowing context or the span itself.
2. Adds basic validations to tag names and start_time.
3. Allows the implementor to retrieve the OpenTracing options he needs to start an span.

Any thoughts?